common/unsafe/src/main/java/org/apache/spark/unsafe/types/ByteArray.java

```diff
@@ -74,4 +74,29 @@ public static long getPrefix(byte[] bytes) {
     }
     return Arrays.copyOfRange(bytes, start, end);
   }
+
+  public static byte[] concat(byte[]... inputs) {
+    // Compute the total length of the result
+    int totalLength = 0;
+    for (int i = 0; i < inputs.length; i++) {
+      if (inputs[i] != null) {
+        totalLength += inputs[i].length;
+      } else {
+        return null;
+      }
+    }
+
+    // Allocate a new byte array, and copy the inputs one by one into it
+    final byte[] result = new byte[totalLength];
+    int offset = 0;
+    for (int i = 0; i < inputs.length; i++) {
+      int len = inputs[i].length;
```

Comment:
null check here too?

Reply:
aha, I see. `UTF8String` seems to need the same null check?

Reply:
nvm. we already have checked here https://github.com/apache/spark/pull/19977/files#diff-6df3223f9826f2b5d1d0c8e29a240ae3R82